internal/venice-client-common/src/main/java/com/linkedin/venice/stats/LongAdderRateGauge.java

```diff
@@ -0,0 +1,53 @@
+package com.linkedin.venice.stats;
+
+import com.linkedin.venice.utils.SystemTime;
+import com.linkedin.venice.utils.Time;
+import io.tehuti.metrics.MetricConfig;
+import io.tehuti.metrics.stats.Gauge;
+import java.util.concurrent.atomic.LongAdder;
+
+
+public class LongAdderRateGauge extends Gauge {
+  private final Time time;
+  private final LongAdder adder = new LongAdder();
+  private long lastMeasurementTime;
+
+  public LongAdderRateGauge() {
+    this(new SystemTime());
+  }
+
+  public LongAdderRateGauge(Time time) {
+    this.time = time;
+    this.lastMeasurementTime = time.getMilliseconds();
+  }
+
+  public void record() {
+    this.adder.increment();
+  }
+
+  public void record(long amount) {
+    this.adder.add(amount);
+  }
+
+  @Override
+  public double measure(MetricConfig config, long currentTimeMs) {
+    return getRate(currentTimeMs);
+  }
+
+  public double getRate() {
+    return getRate(time.getMilliseconds());
+  }
+
+  private double getRate(long currentTimeMs) {
+    double elapsedTimeInSeconds = (double) (currentTimeMs - this.lastMeasurementTime) / Time.MS_PER_SECOND;
+    if (elapsedTimeInSeconds <= 0) {
```

Comment:
can it be < 0 ?

Reply:
No, if we allow == 0 then we get a division by zero.